albumentations/augmentations/transforms.py

```diff
@@ -3320,3 +3321,56 @@ def get_transform_init_args_names(self):
             "mask_fill_value",
             "random_offset",
         )
+
+
+class AugMix(ImageOnlyTransform):
+    """Augmentations mix to Improve Robustness and Uncertainty.
+    Args:
+        image (np.ndarray): Raw input image of shape (h, w, c)
+        severity (int): Severity of underlying augmentation operators.
+        width (int): Width of augmentation chain
+        depth (int): Depth of augmentation chain. -1 enables stochastic depth uniformly
+          from [1, 3]
+        alpha (float): Probability coefficient for Beta and Dirichlet distributions.
+        augmentations (list of augmentations): Augmentations that need to mix and perform.
+    Targets:
+        image
+    Image types:
+        uint8, float32
+
+    Reference:
+    |  https://arxiv.org/abs/1912.02781
+    |  https://github.com/google-research/augmix
+    """
+
+    def __init__(self, width=2, depth=2, alpha=0.5, augmentations=[HorizontalFlip()], always_apply=False, p=0.5):
+        super(AugMix, self).__init__(always_apply, p)
+        self.width = width
+        self.depth = depth
+        self.alpha = alpha
+        self.augmentations = augmentations
+        self.ws = np.float32(np.random.dirichlet([self.alpha] * self.width))
+        self.m = np.float32(np.random.beta(self.alpha, self.alpha))
```

Comment:
Sampling random numbers from numpy in c'tor is a bad practice for two reasons:
1) As we rely on default python random seed, it may have undesirable effect and break reproducible results. You can initialize numpy random generator from seed generated by python random and use it as pseudo-random generator state for `np.random.beta` and `np.random.dirichlet`
2) These weights will be the same for all augmented samples. Is this what was intended? As I understand this could be moved to `get_params`.

Reply:
1.`
self.seed = random.randint(10,10000);
 np.random.seed(self.seed)
`
And then pass seed to init_args_names for reproducibility?
Did you mean something like that?

2. totally agree